### ISSUE TITLE
buildah copy --from container doesn't preserve file ownership, everything is copied as root

### ISSUE THREAD

**Issue by meeuw** [containers/buildah#5587]
**Description**

When using `buildah copy --from container` the file ownership isn't copied (everything is copied as root).

**Steps to reproduce the issue:**
1. `podman run --rm -ti --name meeuw fedora:40`, `touch /meeuw`, `chown 13:13 /meeuw`
2. `WORKING_CONTAINER=$(buildah from fedora:40)`
3. `buildah copy --from meeuw $WORKING_CONTAINER /meeuw /meeuw`
4. `buildah run $WORKING_CONTAINER ls -la /meeuw`

**Describe the results you received:**
`-rw-r--r--. 1 root root 0 Jun 12 17:11 /meeuw`

**Describe the results you expected:**
`-rw-r--r--. 1 13 13 0 Jun 12 17:11 /meeuw`

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.35.4-1.fc40.x86_64
```

**Output of `buildah version`:**

```
Version:         1.35.4
Go Version:      go1.22.2
Image Spec:      1.1.0
Runtime Spec:    1.1.0
CNI Spec:        1.0.0
libcni Version:  
image Version:   5.30.1
Git Commit:      
Built:           Fri May 10 15:40:21 2024
OS/Arch:         linux/amd64
BuildPlatform:   linux/amd64
```

**Output of `podman version` if reporting a `podman build` issue:**

```
Client:       Podman Engine
Version:      5.0.3
API Version:  5.0.3
Go Version:   go1.22.2
Built:        Fri May 10 02:00:00 2024
OS/Arch:      linux/amd64
```

**Output of `cat /etc/*release`:**

```
Fedora release 40 (Forty)
NAME="Fedora Linux"
VERSION="40 (Forty)"
ID=fedora
VERSION_ID=40
VERSION_CODENAME=""
PLATFORM_ID="platform:f40"
PRETTY_NAME="Fedora Linux 40 (Forty)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f40/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=40
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=40
SUPPORT_END=2025-05-13
Fedora release 40 (Forty)
Fedora release 40 (Forty)
```

**Output of `uname -a`:**

```
Linux *** 6.8.11-300.fc40.x86_64 #1 SMP PREEMPT_DYNAMIC Mon May 27 14:53:33 UTC 2024 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
[storage]
driver = "overlay"
runroot = "/run/containers/storage"
graphroot = "/var/lib/containers/storage"
[storage.options]
additionalimagestores = [
]
[storage.options.overlay]
mountopt = "nodev,metacopy=on"
[storage.options.thinpool]
```

**Comment by meeuw** [comment]
I guess this needs a similar fix as #2562

**Comment by rhatdan** [comment]
@nalind PTAL
